### PR TITLE
[Storage] Fix #21494: `az storage blob upload/upload-batch`: Fix `--content-md5` for upload, ignore `--content-md5` for upload-batch

### LINKED ISSUE #21494: az storage blob upload fails with --content-md5 fails on azure-cli 2.34.0
## Describe the bug

**Command Name**
`az storage blob upload ... --content-md5 ...`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
blob_content_md5 must be type bytearray.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 765, in serialize_data
    return self.serialize_type[data_type](data, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 1015, in serialize_bytearray
    return b64encode(attr).decode()
  File "/usr/local/lib/python3.10/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 717, in header
    output = self.serialize_data(data, data_type, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 780, in serialize_data
    raise_with_traceback(
  File "/usr/local/lib/python3.10/site-packages/msrest/exceptions.py", line 51, in raise_with_traceback
    raise error.with_traceback(exc_traceback)
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 765, in serialize_data
    return self.serialize_type[data_type](data, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 1015, in serialize_bytearray
    return b64encode(attr).decode()
  File "/usr/local/lib/python3.10/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
msrest.exceptions.SerializationError: Unable to serialize value: 'y3N9S9wst5SO/myHRqM1uQ==' as type: 'bytearray'., TypeError: a bytes-like object is required, not 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 658, in execute
    raise ex
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 713, in _run_job
    return cmd_copy.exception_handler(ex)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/command_modules/storage/__init__.py", line 380, in new_handler
    first(ex)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/command_modules/storage/__init__.py", line 380, in new_handler
    first(ex)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/command_modules/storage/_exception_handler.py", line 17, in file_related_exception_handler
    raise ex
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 692, in _run_job
    result = cmd_copy(params)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/usr/local/lib/python3.10/site-packages/azure/cli/command_modules/storage/operations/blob.py", line 547, in upload_blob
    response = client.upload_blob(data=stream, length=length, metadata=metadata,
  File "/usr/local/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 83, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/azure/multiapi/storagev2/blob/v2020_10_02/_blob_client.py", line 713, in upload_blob
    return upload_block_blob(**options)
  File "/usr/local/lib/python3.10/site-packages/azure/multiapi/storagev2/blob/v2020_10_02/_upload_helpers.py", line 99, in upload_block_blob
    return client.upload(
  File "/usr/local/lib/python3.10/site-packages/azure/multiapi/storagev2/blob/v2020_10_02/_generated/operations/_block_blob_operations.py", line 195, in upload
    header_parameters['x-ms-blob-content-md5'] = self._serialize.header("blob_content_md5", _blob_content_md5, 'bytearray')
  File "/usr/local/lib/python3.10/site-packages/msrest/serialization.py", line 721, in header
    raise TypeError("{} must be type {}.".format(name, data_type))
TypeError: blob_content_md5 must be type bytearray.

```

## To Reproduce:

- `echo 12765 > /tmp/upload-test-12765.txt`
- `openssl dgst -md5 -binary /tmp/upload-test-12765.txt | base64`
- `az storage blob upload --verbose --container-name stuff-and-things --file /tmp/upload-test-12765.txt --content-md5 y3N9S9wst5SO/myHRqM1uQ== --name dev/test-12765.txt`

## Expected Behavior

Should not fail.
It works on 2.33.1.
On 2.34.0 it works without `--content-md5` option.

## Environment Summary
```
Linux-5.10.76-linuxkit-x86_64-with, Alpine Linux v3.15
Python 3.10.2
Installer: PIP

azure-cli 2.34.0
```

### PR DESCRIPTION
**Pull request by calvinhzy**
**Description**

Fix #21494 . `--content-md5` is required to be bytearray in track2 sdk. It is removed for `az storage blob upload-batch` since it involves a full mapping of all the files to their md5 hash.

**Testing Guide**

echo 12765 > /tmp/upload-test-12765.txt
openssl dgst -md5 -binary /tmp/upload-test-12765.txt | base64
az storage blob upload --verbose --container-name stuff-and-things --file /tmp/upload-test-12765.txt --content-md5 y3N9S9wst5SO/myHRqM1uQ== --name dev/test-12765.txt

**History Notes**

[Storage] BREAKING CHANGE: Fix #21494: `az storage blob upload/upload-batch`: Fix `--content-md5` for upload, ignore `--content-md5` for upload-batch

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py [hunk 1]
```diff
@@ -95,6 +95,29 @@ def register_content_settings_argument(self, settings_class, update, arg_group=N
                        'properties listed below is set, then that property will be cleared.',
                        arg_type=get_three_state_flag())
 
+    def register_content_settings_argument_track2(self, settings_class, update, arg_group=None, guess_from_file=None):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py:98]: Can we add an argument like `include_content_md5` to original func `register_content_settings_argument()` to reduce duplicate code?

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -592,6 +593,73 @@ def validator(cmd, namespace):
     return validator
 
 
+def get_content_setting_validator_track2(settings_class, update, guess_from_file=None, process_md5=False):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:596]: We can add `process_md=False` to the original func `get_content_setting_validator()` to reduce deplication

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py (hunk 1 above)
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py:98]: Same here, we can add `process_md5=False` to the original func `register_conetent_settings_argument()` to reduce duplication

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**